### PR TITLE
[WIP] Make Travis run on a MacOS machine so that iOS artifacts are compiled too

### PR DESCRIPTION
**Pull request by martinbonnin**
I'm opening this PR to investigate Travis support for MacOS. Do not merge yet

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,31 +1,24 @@
...
+  - curl -s "https://get.sdkman.io" | bash
+  - source "$HOME/.sdkman/bin/sdkman-init.sh"
+  - sdk install java 8.0.242.hs-adpt
+  - curl -s https://dl.google.com/android/repository/commandlinetools-mac-6200805_latest.zip -o tools.zip
+  - export ANDROID_SDK_ROOT=~/.android
+  - mkdir $ANDROID_SDK_ROOT
+  - unzip tools.zip -d $ANDROID_SDK_ROOT
+  - export PATH=$ANDROID_SDK_ROOT/tools/bin:$PATH
+  - avdmanager --list
+  - yes | sdkmanager --licenses
```
Comment (tasomaniac) [inline .travis.yml]: Here is what I have in one of my projects. If you put the license, Gradle will automatically install everything necessary 

https://github.com/tasomaniac/OpenLinkWith/blob/master/.travis.yml#L9

Comment (tasomaniac) [reply .travis.yml]: I see that you also need to install system images for emulator

File: build.gradle.kts [hunk 2]
```diff
@@ -300,4 +319,64 @@ fun PublicationContainer.setDefaultPomFields() {
       }
     }
   }
+}
+
+fun publishTasks(repository: String): List<Task> {
```
Comment (tasomaniac) [inline build.gradle.kts]: This function seems to be unused

Comment (martinbonnin) [reply build.gradle.kts]: Not Indeed, thanks! I'll most likely open a new PR with a new branch name as I ended up moving to github actions

**Comment by martinbonnin** [comment]
Superseeded by https://github.com/apollographql/apollo-android/pull/2091/

**Pull request closed without merging**